harness/determined/exec/prep_container.py

```diff
@@ -54,22 +59,125 @@ def trial_prep(info: det.ClusterInfo, cert: certs.Cert) -> None:
         model_def.extractall(path=".")
 
 
-def do_rendezvous(info: det.ClusterInfo, cert: certs.Cert) -> None:
-    # Even though resources_id is not part of the ClusterInfo API, we still depend on it in all
-    # current Determined backends (agents and k8s).
-    resources_id = os.environ.get("DET_RESOURCES_ID")
-    assert resources_id, "Unable to complete rendezvous info without DET_RESOURCES_ID"
-
+def do_rendezvous_rm_provided(
+    info: det.ClusterInfo, cert: certs.Cert, resources_id: str
+) -> "det.RendezvousInfo":
     r = request.get(
         info.master_url,
         path=f"/api/v1/allocations/{info.allocation_id}/resources/{resources_id}/rendezvous",
         cert=cert,
     )
-
     jri = r.json()["rendezvousInfo"]
-    rendezvous_info = det.RendezvousInfo(
-        container_addrs=jri["addresses"], container_rank=jri["rank"]
+    addrs, rank = jri["addresses"], jri["rank"]
+    return det.RendezvousInfo(container_addrs=addrs, container_rank=rank)
+
+
+def do_rendezvous_slurm(
+    info: det.ClusterInfo, cert: certs.Cert, resources_id: str
+) -> "det.RendezvousInfo":
+    rank_str = os.environ.get("SLURM_PROCID")
+    assert rank_str, "Unable to complete rendezvous without SLURM_PROCID"
+    rank = int(rank_str)
+
+    num_peers_str = os.environ.get("SLURM_NPROCS")
+    assert num_peers_str, "Unable to complete rendezvous without SLURM_NPROCS"
+    num_peers = int(num_peers_str)
+
+    rendezvous_ip, resolution_error = None, None
+    for rendezvous_iface in rendezvous_ifaces():
+        try:
+            rendezvous_ip = get_ip_from_interface(rendezvous_iface)
+            break
+        except ValueError as e:
+            resolution_error = e
+    if not rendezvous_ip:
+        raise resolution_error or ValueError("unable to resolve rendezvous ip")
+
+    r = request.post(
```

Comment:
how hard is it to use the generated bindings instead?  This is a new REST API usage, so we should be using the new bindings.  While you're modifying the script, you should also consider migrating the old direct-request code.